docs/register-api-plugins.md

```diff
@@ -0,0 +1,169 @@
+
+At a high level, an API plugin package is one where the main export is an async function that accepts a `ReactionAPI` instance and calls `api.registerPlugin` before returning. In a Reaction API project, your plugin package will be imported and used like this:
+
+```js
+import registerSomePlugin from "some-plugin";
+
+const api = new ReactionAPI();
+
+async function startAPI() {
+  await registerSomePlugin(api);
+  await api.start();
+}
+```
+
+## registerPlugin
+
+A plugin package must export the following function by default:
+
+```js
+export default async function register(app) {
+  // register your plugin here
+  await app.registerPlugin({
+    label: "Plugin Example",
+    name: "plugin-example",
+    version: pkg.version,
+  });
+}
+```
+
+The two keys that every plugin will include are `name` and `label`. `name` must be unique, cannot contain spaces, and identifies your plugin; `label` is the human-readable version of your plugin name, for showing in UIs.
+
+Beyond `name` and `label`, the following standard keys can be included in your `registerPlugin` object:
+
+- `auth`
+- `collections`
+- `contextAdditions`
+- `expressMiddleware`
+- `functionsByType`
+- `graphQL`
+- `i18n`
+- `mutations`
+- `queries`
+- `simpleSchemas`
+
+### auth
+
+Plugins can pass functions in an `auth` object, which are then used to add permission and account information to `context` for each API request.
+
+```js
+auth: {
+  accountByUserId,
+    getHasPermissionFunctionForUser,
+    getShopsUserHasPermissionForFunctionForUser;
+}
+```
+
+- `accountByUserId`: An async function with signature `(context, userId)` that must return an account document for the given `userId`, or `null` if one cannot be found. This will be used to set `context.account` and `context.accountId`.
+- `getHasPermissionFunctionForUser`: An async function with signature `(context)`, which must return an async function that checks permissions for `context.user` and returns a boolean. The signature of the returned function must be `(permissions, shopId)`, where `permissions` is an array of strings and `shopId` is an optional permission filter. The function must return `true` only if the context user has ANY of the permissions.
+- `getShopsUserHasPermissionForFunctionForUser`: An async function with signature `(context)`, which must return an async function that checks shop access for `context.user` and returns a boolean. The signature of the returned function must be `(permission)`, where `permission` is a permission string. The function must return an array of shop IDs for which the context user has the given permission.
+
+### collections
+
+Use this to add new collections to the `context` object. Here is a [guide](how-tos/add-collections-from-plugin.md) to help.
+
+### contextAdditions
+
+A plugin can add properties to context using the `contextAdditions` option when calling `registerPlugin`. They are added before any `preStartup` or `startup` functions are run.
```

Comment:
I think we should mention that we generally recommend against doing this since it's easy to overwrite this should another plugin add something called the same thing. And at minimum they should probably namespace like `myplugin:thingIAmAddding`